src/test/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtilTest.java

```diff
@@ -213,6 +215,75 @@ public void testContainsAnnotationListWithEmptyList() {
             .isFalse();
     }
 
+    @Test
+    public void testHasOverrideNoNode() {
+        final DetailAstImpl ast = new DetailAstImpl();
+        final DetailAstImpl modifiersAst = new DetailAstImpl();
+        modifiersAst.setType(TokenTypes.MODIFIERS);
+        ast.addChild(modifiersAst);
+        final boolean result = AnnotationUtil.hasOverrideAnnotation(ast);
+        assertWithMessage("An empty ast should lead to a false result")
+            .that(result)
+            .isFalse();
+    }
+
+    @Test
+    public void testHasOverrideNull() {
+        final boolean result = AnnotationUtil.hasOverrideAnnotation(null);
+        assertWithMessage("null should lead to a false result")
+            .that(result)
+            .isFalse();
+    }
+
+    @Test
+    public void testHasOverrideOtherAnnotation() {
+        final DetailAstImpl ast = new DetailAstImpl();
+        final DetailAstImpl modifiersAst = create(
+            TokenTypes.MODIFIERS,
+            create(
```

Comment:
such test is most toxic test we can have.
They increase illusion that we do testing/coverage and in same time engineers never respect such tests and update them up to new expectation in new code. So their value in long run is very low.
I presume we do this for pitest, do we have ability to extend tests to be Input based ?

Reply:
Is there some examples of tests, that use inputs for util method?

Such tests are added only for pitest, because they actual functionality is tested by checks UTs, but pitest ignores check for checks.

Utils accepts ASTs, alternative is to parse some input, navigate to some node and pass it to util method, but as for me it looks same as creating AST ourselves

Reply:
if problem come from pitest , we should find a way to resolve it by pitest config.
take a look at: 
https://github.com/checkstyle/checkstyle/blob/aec751dfba46e7ce6eae351cd93fde43f2cf3105/pom.xml#L3272-L3275
we just need to add some tests in pitest-util group for execution.

Historical note: ideally it would be better to run single pitest, so such problem will not exists at all, but unfortunately in this case execution of pitest takes more than 12h, that is why we did a split.

Reply:
done